sql/analyzer/triggers.go

```diff
@@ -355,18 +363,17 @@ func applyTrigger(ctx *sql.Context, a *Analyzer, originalNode, n sql.Node, scope
 		}
 	}
 
-	return transform.NodeWithCtx(n, nil, func(c transform.Context) (sql.Node, transform.TreeIdentity, error) {
+	canApplyTriggerExecutor := func(c transform.Context) bool {
 		// Don't double-apply trigger executors to the bodies of triggers. To avoid this, don't apply the trigger if the
 		// parent is a trigger body.
-		// TODO: this won't work for BEGIN END blocks, stored procedures, etc. For those, we need to examine all ancestors,
-		//  not just the immediate parent. Alternately, we could do something like not walk all children of some node types
-		//  (probably better).
```

Comment:
I think this TODO is still valid. I didn't test the cases it mentioned, but seems like this code is still only looking at immediate parent to decide if we should apply a trigger executor or not, so I think this TODO is worth leaving in. 

Alternatively... if you want to try and test this condition and see if it doesn't work, we could know for sure. If it doesn't work, we could always leave the test as a skipped test (there may be a skipped test for this one somewhere, too). Without testing it though, I think it's better to leave the TODO comment in.